frontends/mit-open/src/common/urls.ts

```diff
@@ -66,6 +66,13 @@ export const login = ({
   return `${LOGIN}?next=${next}`
 }
 
+export const next = () => {
+  const path = window.location.pathname
+  const search = window.location.search
+  const hash = window.location.hash
+  return encodeURIComponent(`${path}${search}${hash}`)
+}
```

Comment:
In general, we should avoid reading from `window.location`. Use
```ts
# Can only be used inside a component.
const location = useLocation()
```
and read from that instead.

**Reason:** Changes to global variables (like `window.location`) won't cause react to re-render[^1]. So for example, suppose the login link (which is styled as a button) reads its `href` from `window.location`. Then when the user routes within the app, react will have an out-of-date `href` one the login button. 

[^1]: In the case of `window.location`, this depends on how it changes. If you change `window.location` by directly assigning to `window.location.href`, then the whole page reloads, so ... yeah, in that case, react would re-render. React router uses the [history API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) to change `window.location` without doing a full refresh.

When you use `const location = useLocation()`, the `useLocation` hook gets its value from a [React Context](https://react.dev/learn/passing-data-deeply-with-context). Then related routing hooks (like `useNavigate` and `useSearchParams`) change the value of the context, which causes react to re-render.